src/main/core.js

```diff
@@ -319,7 +327,8 @@ async function formatWithCursor(originalText, originalOptions) {
 
   if (
     (options.rangeStart >= options.rangeEnd && text !== "") ||
-    (options.requirePragma && !(await hasPragma(text, options)))
+    (options.requirePragma && !(await hasPragma(text, options))) ||
+    (options.checkIgnorePragma && (await hasIgnorePragma(text, options)))
```

Comment:
1. Should we enable this feature if `--require-pragma` used?
2. What will happen if two flags are used together?

Reply:
Currently, the two options act independently and look at different pragmas.
- If you have `requirePragma` enabled, you need `@prettier` in order to format.
- If you have `checkIgnorePragma enabled, you need to not have `@noprettier` to format.

If both options are enabled, you only format files that have `@prettier` and don't have `@noprettier`. From the perspective of both features limiting which files get formatted, neither really has power over the other; both can exclude files for different reasons.

That's my take on it anyway, I don't feel too strongly. That's the way it landed in implementing it, and other behaviors would likely require additional code.

I could add tests if we want to formalize the behavior.